src/forms/CalendarComponent.jsx

```diff
@@ -38,25 +36,30 @@ class CalendarComponent extends React.Component {
 			defaultDate: this.props.value
 		};
 
-		Object.assign(options, this.props.opts);
+		Object.assign(options, this.props.datepickerOptions);
 		this.flatpickr = new Flatpickr(this.inputEl, options);
 	}
 
 	componentWillUnmount() {
 		this.flatpickr && this.flatpickr.destroy();
 	}
 
+	// replaces updateFlatpickr
+	// if we receive a new value from parent, update Flatpickr
+	componentWillReceiveProps(newProps) {
+		this.flatpickr.setDate(newProps.value);
```

Comment:
`componentWillUnmount` and the old `updateFlatpickr` method check for a falsey `this.flatpickr` - should probably do the same here unless there's a guarantee that `this.flatpickr` is always available, in which case the falsey check can be removed from `componentWillUnmount`